libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/service/MapboxTripService.kt

```diff
@@ -30,9 +28,25 @@ internal class MapboxTripService(
 
     companion object {
 
-        private var notificationDataChannel = Channel<MapboxNotificationData>(1)
-        internal fun getNotificationDataChannel(): ReceiveChannel<MapboxNotificationData> =
-            notificationDataChannel
+        private var currentTripNotification: TripNotification? = null
+
+        private val notificationDataObservers = CopyOnWriteArraySet<NotificationDataObserver>()
+
+        internal fun registerOneTimeNotificationDataObserver(observer: NotificationDataObserver) {
+            currentTripNotification?.let { tripNotification ->
+                val notificationData = MapboxNotificationData(
+                    tripNotification.getNotificationId(),
+                    tripNotification.getNotification(),
+                )
+                observer.onNotificationUpdated(notificationData)
+                return
+            }
+            notificationDataObservers.add(observer)
+        }
+
+        internal fun unregisterOneTimeNotificationDataObserver(observer: NotificationDataObserver) {
+            notificationDataObservers.remove(observer)
```

Comment:
```suggestion
            if (!notificationDataObservers.isEmpty()) {
                notificationDataObservers.remove(observer)
            }
```

Reply:
I see that `CopyOnWriteArraySet.remove` basically does nothing without any extra overhead if the list is empty already, so I don't see why that check is necessary.